src/Container/Exceptions/CannotAutowireException.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Container\Exceptions;
+
+use Exception;
+use Tempest\Container\ContainerLog;
+
+final class CannotAutowireException extends Exception
+{
+    public function __construct(ContainerLog $containerLog)
+    {
+        $stack = $containerLog->getStack();
+
+        $firstContext = $stack[array_key_first($stack)];
+        $lastContext = $stack[array_key_last($stack)];
+
+        $message = PHP_EOL . PHP_EOL . "Cannot autowire {$firstContext->getName()} because {$lastContext->getName()} cannot be resolved" . PHP_EOL;
+
+        $i = 0;
+
+        foreach ($stack as $currentContext) {
+            $pipe = match ($i) {
+                0 => '┌──',
+                count($stack) - 1 => '└──',
+                default => '├──',
+            };
+
+            $message .= PHP_EOL . "\t{$pipe} " . $currentContext;
+
+            $i++;
+        }
+
+        $currentDependency = $lastContext->currentDependency();
+        $currentDependencyName = (string) $currentDependency;
+        $firstPart = explode($currentDependencyName, (string) $lastContext)[0] ?? null;
+        $fillerLines = str_repeat(' ', strlen($firstPart) + 3);
```

Comment:
Shouldn't we rename this to something like `$fillerSpaces` for clarity?

Reply:
Definitely! I really don't think this is the greatest code btw. But it works for now, and is contained in one class, so I'm fine with it.

It would be interesting to build some kind of "highlighting component", but that's not for today 😁